lib/rubocop/cop/rspec/repeated_example_group_description.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated example group descriptions.
+      #
+      # @example
+      #
+      #    # bad
+      #    describe 'cool feature' do
+      #      # example group
+      #    end
+      #
+      #    describe 'cool feature' do
+      #      # example group
+      #    end
+      #
+      #    # bad
+      #    context 'when case x' do
+      #      # example group
+      #    end
+      #
+      #    describe 'when case x' do
+      #      # example group
+      #    end
+      #
+      #    # bad
+      #    RSpec.describe 'Animal', 'dog' do
+      #      # example group
+      #    end
+      #
+      #    RSpec.describe 'Animal', 'cat' do
+      #      # example group
+      #    end
+      #
+      #    # good
+      #    describe 'cool feature' do
+      #      # example group
+      #    end
+      #
+      #    describe 'another cool feature' do
+      #      # example group
+      #    end
+      #
+      #    # good
+      #    context 'when case x' do
+      #      # example group
+      #    end
+      #
+      #    context 'when another case' do
+      #      # example group
+      #    end
+      #
+      class RepeatedExampleGroupDescription < Cop
+        MSG = 'Repeated %<group>s block description on line(s) %<loc>s'
+
+        def_node_matcher :several_groups?, <<-PATTERN
+          (begin (block (send _ {#{ExampleGroups::ALL.node_pattern}} ...) ...)+ )
+        PATTERN
+
+        def_node_matcher :extract_doc_string, '(block (send _ _ $str ...) ...)'
```

Comment:
Are you sure that `extract_` adds some value in the method name?

Reply:
@pirj just tried to make method names self explainable